From 88cfe105614a576580bc3f36ee1f0d91cd37b781 Mon Sep 17 00:00:00 2001
From: Daniel Leung <danielcp@gmail.com>
Date: Thu, 5 Mar 2020 17:20:25 -0800
Subject: [PATCH 5/8] gdb: xtensa: don't error out when registers cannot be
 read

On Xtensa, some privileged registers cannot be read using 'p'
packet. So keep these marked as unavailable so GDB can still
fetch other registers instead of stopping at error.

Signed-off-by: Daniel Leung <danielcp@gmail.com>
---
 gdb/remote.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/gdb/remote.c b/gdb/remote.c
index 36136e3e3e..c8c5b32c74 100644
--- a/gdb/remote.c
+++ b/gdb/remote.c
@@ -7933,6 +7933,7 @@ remote_target::fetch_register_using_p (struct regcache *regcache,
   char *buf, *p;
   gdb_byte *regp = (gdb_byte *) alloca (register_size (gdbarch, reg->regnum));
   int i;
+  enum bfd_architecture bfd_arch = gdbarch_bfd_arch_info (gdbarch)->arch;
 
   if (packet_support (PACKET_p) == PACKET_DISABLE)
     return 0;
@@ -7956,6 +7957,15 @@ remote_target::fetch_register_using_p (struct regcache *regcache,
     case PACKET_UNKNOWN:
       return 0;
     case PACKET_ERROR:
+      /* On Xtensa, some privileged registers cannot be read using 'p'
+         packet. So keep these marked as unavailable so GDB can still
+         fetch other registers instead of stopping at error.  */
+      if (bfd_arch == bfd_arch_xtensa)
+        {
+          regcache->raw_supply (reg->regnum, NULL);
+          return 1;
+        }
+
       error (_("Could not fetch register \"%s\"; remote failure reply '%s'"),
 	     gdbarch_register_name (regcache->arch (), 
 				    reg->regnum), 
-- 
2.24.1

